ui/src/views/Roles/ViewRoles/index.jsx

```diff
@@ -16,10 +17,18 @@ import HelpView from '../../../components/HelpView';
 }))
 export default class ViewRoles extends PureComponent {
   state = {
-    roleSearch: '',
+    roleSearch: this.props.history.location.search
+      ? parse(this.props.history.location.search.slice(1)).roleSearch
+      : '',
   };
 
   handleRoleSearchSubmit = roleSearch => {
+    if (roleSearch !== this.state.roleSearch) {
+      this.props.history.push(
+        roleSearch.length > 0 ? `?${stringify({ roleSearch })}` : '/auth/roles'
+      );
```

Comment:
I think you should be able to use (untested)

```jsx
const { location, history } = this.props;
const query = parse(location.search.slice(1));

if (query.roleSearch !== roleSearch) {
  query.roleSearch = roleSearch;

  history.push({
    search: stringify(query, { addQueryPrefix: true }),
  });
}
```

This would make sure that if we decide to add more queryies in the future, they won't be removed when a user submits a search. Let's do the same thing in the other view as well.

Reply:
Looks great! Let me have a try, thanks!